### PR TITLE
MULE-9475: Allow config level components in the Extensions API

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/ExtensionManagerAdapter.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * @since 4.0
+ */
+public interface ExtensionManagerAdapter extends ExtensionManager
+{
+
+    /**
+     * Locates and returns the {@link ConfigurationProvider} which would serve
+     * an invokation to the {@link #getConfiguration(ExtensionModel, Object)} method.
+     * <p>
+     * This means that the returned provided will be located using the same set of rules
```
Comment (afelisatti) [inline modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/ExtensionManagerAdapter.java]: provider?

File: modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/ExtensionManagerAdapter.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+     * as the aforementioned method
+     *
+     * @param extensionModel the {@link ExtensionModel} for which a configuration is wanted
+     * @param <C>            the generic type of the configuration instance that the provider returns
+     * @return an {@link Optional} {@link ConfigurationProvider}
+     */
+    <C> Optional<ConfigurationProvider<C>> getConfigurationProvider(ExtensionModel extensionModel);
+
+    /**
+     * Locates and returns the {@link ConfigurationProvider} which would serve an invokation to the
```
Comment (afelisatti) [inline modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/ExtensionManagerAdapter.java]: invocation

File: modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/ExtensionManagerAdapter.java (hunk 1 above)
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/ExtensionManagerAdapter.java]: fixed

File: modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/ExtensionManagerAdapter.java (hunk 2 above)
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/ExtensionManagerAdapter.java]: fixed

File: modules/extensions-support/src/test/java/org/mule/module/extension/internal/runtime/config/DynamicConfigurationProviderTestCase.java [hunk 3]
```diff
@@ -63,8 +63,10 @@ public void before() throws Exception
         when(configurationModel.getConfigurationFactory().newInstance()).thenAnswer(invocation -> MODULE_CLASS.newInstance());
         when(configurationModel.getModelProperty(any())).thenReturn(Optional.empty());
         when(configurationModel.getInterceptorFactories()).thenReturn(ImmutableList.of());
+        when(configurationModel.getOperationModels()).thenReturn(ImmutableList.of());
         when(configurationModel.getExtensionModel().getOperationModels()).thenReturn(ImmutableList.of());
 
+
```
Comment (afelisatti) [inline modules/extensions-support/src/test/java/org/mule/module/extension/internal/runtime/config/DynamicConfigurationProviderTestCase.java]: extra line

Comment (marianogonzalez) [inline modules/extensions-support/src/test/java/org/mule/module/extension/internal/runtime/config/DynamicConfigurationProviderTestCase.java]: fixed

**Pull request merged**